Code/Framework/AzTest/AzTest/Platform/Windows/ScopedAutoTempDirectory_Windows.cpp

```diff
@@ -8,57 +8,41 @@
 
 #include <AzCore/PlatformIncl.h>
 
+#include <AzCore/std/string/conversions.h>
 #include <AzCore/IO/Path/Path.h>
 #include <AzCore/std/functional.h>
 #include <AzCore/IO/SystemFile.h>
-#include <AzTest/Platform.h>
+#include <AzTest/Utils.h>
 
-namespace AZ
+namespace AZ::Test
 {
-    namespace Test
+    ScopedAutoTempDirectory::ScopedAutoTempDirectory()
     {
-        ScopedAutoTempDirectory::ScopedAutoTempDirectory()
-        {
-            constexpr const DWORD bufferSize = static_cast<DWORD>(AZ::IO::MaxPathLength);
+        using UuidString = AZStd::fixed_string<AZ::Uuid::MaxStringBuffer>;
+        constexpr DWORD bufferSize = static_cast<DWORD>(AZ::IO::MaxPathLength);
 
-            char tempDir[bufferSize] = {0};
-            GetTempPathA(bufferSize, tempDir);
+        wchar_t tempDirW[AZ::IO::MaxPathLength]{};
+        GetTempPathW(bufferSize, tempDirW);
 
-            char workingTempPathBuffer[bufferSize] = {'\0'};
+        AZ::IO::FixedMaxPath tempDirectoryRoot;
+        AZStd::to_string(tempDirectoryRoot.Native(), tempDirW);
 
-            int maxAttempts = 2000; // Prevent an infinite loop by setting an arbitrary maximum attempts at finding an available temp folder name
-            while (maxAttempts > 0)
+        constexpr int MaxAttempts = 255;
+        for (int i = 0; i < MaxAttempts; ++i)
+        {
+            AZ::IO::FixedMaxPath testPath = tempDirectoryRoot /
+                AZ::IO::FixedMaxPathString::format("UnitTest-%s",
+                    AZ::Uuid::CreateRandom().ToString<UuidString>().c_str());
```

Comment:
Almost feels like a mutex is needed somewhere during the random generation and creation to be extra safe, but probably not

Reply:
The [Uuid::CreateRandom](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/Uuid.cpp#L249-L252) function uses the [Sfmt::Rand32](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/Sfmt.cpp#L516-L521) function which already provides mutex protection against the generation of the random number.

Reply:
Yeah, i was thinking of the whole create random -> create folder block (sorry didnt highlight the whole section). But random generation is done through a sequence anyways, so probably would never be an issue, so it wouldnt really matter anyways

Reply:
Well there are no shared variables that could run on multiple threads.

The temporary directory creation is in a constructor, which can only occur on a single thread.